addons/spreadsheet_account/models/account.py

```diff
@@ -34,7 +35,7 @@ def _get_date_period_boundaries(self, date_period, company):
             start = date(year, first_month, 1)
             end = start + relativedelta(months=3, days=-1)
         elif period_type == "day":
-            fiscal_day = company.fiscalyear_last_day
+            fiscal_day = company._get_fiscalyear_last_day(year)
```

Comment:
All of that code is wrong. In any case, we need to use the `compute_fiscalyear_dates` method on `res.company`. This method is there to handle such issues and to manage the custom fiscal years.

Reply:
> All of that code is wrong. In any case, we need to use the `compute_fiscalyear_dates` method on `res.company`. This method is there to handle such issues and to manage the custom fiscal years.

Can you please explain your suggestion deeply what exactly need to do because I am confused!

Reply:
@uso-odoo From my understanding, something like that should do the job:
```py
from odoo.tools.date_utils import get_month, get_quarter

@api.model
def _get_date_period_boundaries(self, date_period, company):
    period_type = date_period["range_type"]
    day = date_period['day']
    month = date_period['month']
    year = date_period['year']
    quarter = date_period.get("quarter")
    current_date = date(year, month, day)
    if period_type == "year":
        # If you want to take the accounting fiscal year:
        fiscalyear_dates = company.compute_fiscalyear_dates(current_date)
        start = fiscalyear_dates['date_from']
        end = fiscalyear_dates['date_to']
    elif period_type == "month":
        start, end = get_month(current_date.day, current_date.month)
    elif period_type == "quarter":
        start, end = get_quarter(current_date.day, current_date.month)
    elif period_type == "day":
        fiscalyear_dates = company.compute_fiscalyear_dates(current_date)
        start = fiscalyear_dates['date_from']
    return start, end
```

Reply:
> @uso-odoo From my understanding, something like that should do the job:
> 
> ```python
> from odoo.tools.date_utils import get_month, get_quarter
> 
> @api.model
> def _get_date_period_boundaries(self, date_period, company):
>     period_type = date_period["range_type"]
>     day = date_period['day']
>     month = date_period['month']
>     year = date_period['year']
>     quarter = date_period.get("quarter")
>     current_date = date(year, month, day)
>     if period_type == "year":
>         fiscalyear_dates = company.compute_fiscalyear_dates(current_date)
>         start = fiscalyear_dates['date_from']
>         end = fiscalyear_dates['date_to']
>     elif period_type == "month":
>         start, end = get_month(current_date.day, current_date.month)
>     elif period_type == "quarter":
>         start, end = get_quarter(current_date.day, current_date.month)
>     elif period_type == "day":
>         fiscalyear_dates = company.compute_fiscalyear_dates(current_date)
>         start = fiscalyear_dates['date_from']
>     return start, end
> ```

Hello @smetl 

Here is my explanation about why we can not use `compute_fiscalyear_dates`:-

- If we push our solution by using the `compute_fiscalyear_dates` method, and if the 'account_accountant` module is not installed (only `account` is installed) during that time these 2 tests failed `test_first_fiscal_day_date_period` and `test_shifted_fiscal_year_date_period` because it will run `compute_fiscalyear_dates` from `account` module where (For Example:- if our `current_date = 04-02-2023`) then the logic is it will take our current date's year and statically it will return start date as a 1st January of the current date's year and end date as a 31st December of the current date's year!
- If you still want to use the `compute_fiscalyear_dates` method in `_get_date_period_boundaries` then we need to modify `account` module's `compute_fiscalyear_dates` method so do let me know your suggestion on it!
- If not then this solution solved this issue kindly check it and let me know further!

Thanks!